### ISSUE TITLE
Example swagger-codegen-maven-plugin/examples/java-client.xml is out of date

### ISSUE THREAD

**Issue by Trii** [swagger-api/swagger-codegen#3261]
Example dependencies from [swagger-codegen-maven-plugin/examples/java-client.xml](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen-maven-plugin/examples/java-client.xml) do not work correctly.

I had to use the generation webservice to extract a pom with the proper dependencies for jersey2

They should be changed to something like this if they are indeed the best practices:

``` xml
<!-- HTTP client: jersey-client -->
    <dependency>
      <groupId>org.glassfish.jersey.core</groupId>
      <artifactId>jersey-client</artifactId>
      <version>${jersey-version}</version>
    </dependency>
    <dependency>
      <groupId>org.glassfish.jersey.media</groupId>
      <artifactId>jersey-media-multipart</artifactId>
      <version>${jersey-version}</version>
    </dependency>
    <dependency>
      <groupId>org.glassfish.jersey.media</groupId>
      <artifactId>jersey-media-json-jackson</artifactId>
      <version>2.22.1</version>
    </dependency>

    <!-- JSON processing: jackson -->
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-core</artifactId>
      <version>${jackson-version}</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-annotations</artifactId>
      <version>${jackson-version}</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>${jackson-version}</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.datatype</groupId>
      <artifactId>jackson-datatype-joda</artifactId>
      <version>2.1.5</version>
    </dependency>
    <dependency>
      <groupId>joda-time</groupId>
      <artifactId>joda-time</artifactId>
      <version>${jodatime-version}</version>
    </dependency>

    <!-- Base64 encoding that works in both JVM and Android -->
    <dependency>
      <groupId>com.brsanthu</groupId>
      <artifactId>migbase64</artifactId>
      <version>2.2</version>
    </dependency>

    <!-- test dependencies -->
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>${junit-version}</version>
      <scope>test</scope>
    </dependency>
  </dependencies>
  <properties>
    <swagger-core-version>1.5.8</swagger-core-version>
    <jersey-version>2.22.2</jersey-version>
    <jackson-version>2.7.0</jackson-version>
    <jodatime-version>2.9.3</jodatime-version>
    <maven-plugin-version>1.0.0</maven-plugin-version>
    <junit-version>4.12</junit-version>
  </properties>
```

**Comment by wing328** [comment]
@Trii thanks for reporting the issue. May I know if you've cycle to contribute a fix?

**Comment by Trii** [comment]
I can attempt it but I can't guarantee that the version choices are optimal since I'm new to this library.

**Comment by wing328** [comment]
That's ok. The community and the core team can help review the change :)

Let us know if you need help submitting a PR.

**Comment by frantuma** [comment]
fixed in #8448
